test/system/src/test/java/io/pravega/test/system/MultiReaderWriterWithFailOverTest.java

```diff
@@ -0,0 +1,424 @@
+        eventWriteCount1 = eventData;
+        eventReadCount1 = eventReadCount;
+
+        log.info("Read count without any failure {}", eventReadCount1.get());
+        log.info("Write count without any failure {}", eventWriteCount1.get());
+
+        //scale after some random time
+        Thread.sleep(new Random().nextInt(50000));
+
+        //Scale down SSS instances to 2
+        segmentStoreInstance.scaleService(2, true);
+        log.info("Scaling down SSS instances from 3 to 2");
+        eventWriteCount2 = eventData;
+        eventReadCount2 = eventReadCount;
+
+        log.info("Read count with SSS failover {}", eventReadCount2.get());
+        log.info("Write count with SSS failover {}", eventWriteCount2.get());
+
+        //ensure reads are happening
+        assertTrue(eventReadCount2.get() > eventReadCount1.get());
+        //ensure writes are happening
+        assertTrue(eventWriteCount2.get() > eventWriteCount1.get());
+
+        //scale after some random time
+        Thread.sleep(new Random().nextInt(50000));
+
+        //Scale down controller instances to 2
+        controllerInstance.scaleService(2, true);
+        log.info("Scaling down controller instances from 3 to 2");
+        List<URI> conURIs = controllerInstance.getServiceDetails();
+        // Fetch all the RPC endpoints and construct the client URIs.
+        final List<String> uris = conURIs.stream().filter(uri -> uri.getPort() == 9092).map(URI::getAuthority)
+                .collect(Collectors.toList());
+        controllerURIDirect = URI.create("tcp://" + String.join(",", uris));
+        log.info("Controller Service direct URI: {}", controllerURIDirect);
+        eventWriteCount3 = eventData;
+        eventReadCount3 = eventReadCount;
+
+        log.info("Read count with controller failover {}", eventReadCount3.get());
+        log.info("Write count with controller failover {}", eventWriteCount3.get());
+
+        //ensure reads are happening
+        assertTrue(eventReadCount3.get() > eventReadCount2.get());
+        //ensure writes are happening
+        assertTrue(eventWriteCount3.get() > eventWriteCount2.get());
+
+        //scale after some random time
+        Thread.sleep(new Random().nextInt(50000));
+
+        //Scale down SSS, controller to 1 instance each.
+        segmentStoreInstance.scaleService(1, true);
+        controllerInstance.scaleService(1, true);
+        log.info("Scaling down  to 1 controller, 1 SSS instance");
+        List<URI> conURIs1 = controllerInstance.getServiceDetails();
+        // Fetch all the RPC endpoints and construct the client URIs.
+        final List<String> uris1 = conURIs1.stream().filter(uri -> uri.getPort() == 9092).map(URI::getAuthority)
+                .collect(Collectors.toList());
+        controllerURIDirect = URI.create("tcp://" + String.join(",", uris1));
+        log.info("Controller Service direct URI: {}", controllerURIDirect);
+        eventWriteCount4 = eventData;
```

Comment:
eventWriteCount4 should be of type long, please modify all other variables used to take a snapshot to long.